drivers/usb/device/usb_dc_rpi_pico.c

```diff
@@ -336,6 +343,20 @@ static void udc_rpi_isr(const void *arg)
 		k_msgq_put(&usb_dc_msgq, &msg, K_NO_WAIT);
 	}
 
+	if (status & USB_INTS_VBUS_DETECT_BITS) {
+		handled |= USB_INTS_VBUS_DETECT_BITS;
+		hw_clear_alias(usb_hw)->sie_status = USB_SIE_STATUS_VBUS_DETECTED_BITS;
+#if IS_ENABLED(CONFIG_USB_DC_RPI_PICO_HW_VBUS_DETECT)
```

Comment:
It is preferred to use `if (IS_ENABLED(CONFIG_USB_DC_RPI_PICO_HW_VBUS_DETECT))` instead of `#ifdef` to have the code always compile (and be discarded by the compiler if config is not set) in order to avoid "compilation breaks only with specific kconfigs" issues.